core/src/main/codegen/templates/Parser.jj

```diff
@@ -6176,6 +6176,21 @@ SqlNode BuiltinFunctionCall() :
         <RPAREN> {
             return SqlStdOperatorTable.EXTRACT.createCall(s.end(this), args);
         }
+    |
+        <DATE> { s = span(); }
+        <LPAREN>
+        e = Expression(ExprContext.ACCEPT_SUB_QUERY) {
+            args.add(e);
```

Comment:
Args isn't defined, so we get a null pointer exception here. I've changed this to `args = startList(e);`, which initializes the list.

There seems to be a few minor errors with the quotation in the expected return values, but it seems very minor.